src/data/battle-anims.ts

```diff
@@ -511,6 +517,21 @@ export function initMoveAnim(scene: BattleScene, move: Moves): Promise<void> {
   });
 }
 
+export function initEncounterAnims(scene: BattleScene): Promise<void> {
+  return new Promise(resolve => {
+    const encounterAnimNames = Utils.getEnumKeys(EncounterAnim);
+    const encounterAnimIds = Utils.getEnumValues(EncounterAnim);
+    const encounterAnimFetches = [];
+    for (let ea = 0; ea < encounterAnimIds.length; ea++) {
+      const encounterAnimId = encounterAnimIds[ea];
+      encounterAnimFetches.push(scene.cachedFetch(`./battle-anims/encounter-${encounterAnimNames[ea].toLowerCase().replace(/\_/g, "-")}.json`)
+        .then(response => response.json())
+        .then(cas => encounterAnims.set(encounterAnimId, new AnimConfig(cas))));
+    }
+    Promise.allSettled(encounterAnimFetches).then(() => resolve());
+  });
+}
+
```

Comment:
Please try to use `async/await` as much as possible

```suggestion
export async function initEncounterAnims(scene: BattleScene): Promise<void> {
  const encounterAnimNames = Utils.getEnumKeys(EncounterAnim);
  const encounterAnimIds = Utils.getEnumValues(EncounterAnim);
  const encounterAnimFetches = [];

  for (let ea = 0; ea < encounterAnimIds.length; ea++) {
    const encounterAnimId = encounterAnimIds[ea];
    const fetchPromise = scene.cachedFetch(`./battle-anims/encounter-${encounterAnimNames[ea].toLowerCase().replace(/\_/g, "-")}.json`)
      .then(response => response.json())
      .then(cas => encounterAnims.set(encounterAnimId, new AnimConfig(cas)));
    encounterAnimFetches.push(fetchPromise);
  }

  await Promise.allSettled(encounterAnimFetches);
}

```
<sup> CODE NOT TESTED!</sup>

Reply:
Also add some jsdoc